tree/treeplayer/inc/ROOT/TDFNodes.hxx

```diff
@@ -105,6 +130,7 @@ class TLoopManager : public std::enable_shared_from_this<TLoopManager> {
    std::string fToJit;        ///< string containing all `BuildAndBook` actions that should be jitted before running
    const std::unique_ptr<TDataSource> fDataSource; ///< Owning pointer to a data-source object. Null if no data-source
    ColumnNames_t fDefinedDataSourceColumns; ///< List of data-source columns that have been `Define`d so far
+   std::vector<TLoopCallback> fCallbacks; ///< Registered callbacks invoked once per event per slot
```

Comment:
If it is invoke once per event per slot (so once per event), what is the semantic difference with an (unfiltered) part of the functional chain applied to each events?

Reply:
Humm ... Actually maybe the comment is obsolete as it seems to not match well with the "Never call callbacks concurrently, only ever run on slot 0" ....

Reply:
Yes the comment there is wrong. I made several changes to the behavior of the feature after your comments as well as other people's. New revised version is coming in a minute :)